doc/API/util/piped_process.md

```diff
@@ -0,0 +1,25 @@
+# `src/util/piped_process.{cpp, h}`
+
+To utilise the `piped_process` API for interprocess communication with any binary:
+
+* You need to initialise it by calling the construct `piped_processt("binary with args", message_handler)`.
+  * A `message_handlert` should usually be present down from the call stack at 
+    the place you want to call `piped_process` from.
+  * If IPC fails during child process creation, you will get a `system_exceptiont`.
+  * If the `binary command` does not correspond to a binary in the `$PATH` or is
+    not a path to a binary itself, you'll read a string `Launching <xyz> failed with error: <error>`
+    when you attempt to `receive()` output from it.
+* To close the underlying child process, you need a call to the destructor - `~piped_processt()`.
```

Comment:
I think I would rephrase this to be something like:
"The child process is automatically killed with SIGTERM when the destructor for the piped_processt is called."